users/userRouter.js

```diff
@@ -1,47 +1,140 @@
-const express = 'express';
+const express = require("express");
 
-const router = express.Router();
+const Users = require("./userDb.js");
 
-router.post('/', (req, res) => {
+const router = express.Router();
 
+router.post("/", validateUser, (req, res) => {
+  const user = req.body;
+  Users.insert(user)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "cannot post this user idk"
+      });
+    });
 });
 
-router.post('/:id/posts', (req, res) => {
-
+router.post("/:id/posts", validateUserId, validatePost, (req, res) => {
+  const user = req.body;
+  Users.insert(user)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "what do i even put here?"
+      });
+    });
 });
 
-router.get('/', (req, res) => {
-
+router.get("/", (req, res) => {
+  Users.get()
+    .then(users => {
+      res.status(200).json(users);
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: "am i getting it?"
+      });
+    });
 });
 
-router.get('/:id', (req, res) => {
-
+router.get("/:id", validateUserId, (req, res) => {
+  const user = req.user;
+  res.status(200).json(user);
 });
 
-router.get('/:id/posts', (req, res) => {
-
+router.get("/:id/posts", validateUserId, validatePost, (req, res) => {
+  const { id } = req.params;
```

Comment:
validatePost doesn't belong here, it should probably only be on a posts POST request